### ISSUE TITLE
[java] UnusedImport false positive for star import

### ISSUE THREAD

**Issue by krichter722** [pmd/pmd#2050]
**Affects PMD Version:** 6.18.0

**Rule:** [UnusedImports](https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#unusedimports)

**Description:**
After importing `org.some.package.*` I'm able to use all classes `org.some.package.A`, `org.some.package.B`, etc. with their class names `A`, `B`, etc. Thus the failure demonstrated in the code sample is a false positive.

**Code Sample demonstrating the issue:**

```java
package de.richtercloud.pmd.star.imports.unused;

import org.apache.commons.text.*;

public class SomeClass {

    public void someMethod() {
        WordUtils.capitalize("abc");
        CaseUtils.toCamelCase("abc", true);
    }
}
```

causes a violation if 

```
<?xml version="1.0"?>
<ruleset name="Custom Rules"
         xmlns="http://pmd.sourceforge.net/ruleset/2.0.0"
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://pmd.sourceforge.net/ruleset/2.0.0 https://pmd.sourceforge.io/ruleset_2_0_0.xsd">
    <rule ref="category/java/bestpractices.xml/UnusedImports" />
</ruleset>
```

is specified.

**Running PMD through:** *Maven*

```
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0"
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
    <modelVersion>4.0.0</modelVersion>

    <groupId>de.richtercloud</groupId>
    <artifactId>pmd-star-imports-unused</artifactId>
    <version>1.0-SNAPSHOT</version>

    <properties>    <maven.compiler.source>1.8</maven.compiler.source>
        <maven.compiler.target>1.8</maven.compiler.target>
        <pmd.version>6.19.0-SNAPSHOT</pmd.version>
    </properties>

    <dependencies>
        <dependency>
            <groupId>org.apache.commons</groupId>
            <artifactId>commons-text</artifactId>
            <version>1.8</version>
        </dependency>
    </dependencies>

    <build>
        <plugins>
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-pmd-plugin</artifactId>
                <version>3.12.0</version>
                <executions>
                    <execution>
                        <id>pmd</id>
                        <phase>verify</phase>
                        <goals>
                            <goal>check</goal>
                        </goals>
                    </execution>
                </executions>
                <configuration>
                    <rulesets>
                        <ruleset>pmd-ruleset.xml</ruleset>
                    </rulesets>
                    <printFailingErrors>true</printFailingErrors>
                    <includeTests>true</includeTests>
                </configuration>
                <dependencies>
                    <dependency>
                        <groupId>net.sourceforge.pmd</groupId>
                        <artifactId>pmd-core</artifactId>
                        <version>${pmd.version}</version>
                    </dependency>
                    <dependency>
                        <groupId>net.sourceforge.pmd</groupId>
                        <artifactId>pmd-java</artifactId>
                        <version>${pmd.version}</version>
                    </dependency>
                </dependencies>
            </plugin>
        </plugins>
    </build>
</project>
```

**Comment by adangel** [comment]
Thanks for the report. Yes, that's a false positive. We obviously don't detect method calls such as `WordUtils.capitalize("abc");` correctly.

**Comment by jsotuyod** [comment]
This is a duplicate of #2016, we will follow it there.
